cmd/system-probe/loader.go

```diff
@@ -31,17 +33,26 @@ func (l *Loader) Register(cfg *config.AgentConfig, httpMux *http.ServeMux, facto
 			continue
 		}
 
+		// In case a module failed to be started, do not make the whole `system-probe` abort.
+		// Let `system-probe` run the other modules.
 		if err != nil {
-			return errors.Wrapf(err, "new module `%s` error", factory.Name)
+			log.Errorf("new module `%s` error: %w", factory.Name, err)
+			continue
 		}
 
 		if err = module.Register(httpMux); err != nil {
-			return errors.Wrapf(err, "error registering HTTP endpoints for module `%s` error", factory.Name)
+			log.Errorf("error registering HTTP endpoints for module `%s` error: %w", factory.Name, err)
+			continue
 		}
 
 		l.modules[factory.Name] = module
 
 		log.Infof("module: %s started", factory.Name)
+		atLeastOneModuleLoaded = true
+	}
+
+	if !atLeastOneModuleLoaded {
```

Comment:
You could use `if len(l.modules) > 0` here if you wanted.

Reply:
Indeed. Fixed!